tagcompare/logger.py

```diff
@@ -0,0 +1,56 @@
+import os
+import logging
+
+import settings
+
+
+class Logger(object):
+    def __init__(self, name, directory=None, writefile=False):
+        if directory and not writefile:
+            raise ValueError("directory was specified when writefile is False!")
+
+        self.__directory = directory
+        if writefile:
+            if not directory:
+                self.__directory = os.path.join(
+                    os.path.abspath(os.path.dirname(__file__)), settings.LOG_DIR)
+            if not os.path.exists(self.__directory):
+                raise IOError("directory doesn't exist at {}".format(self.__directory))
+
+        self.__name = name.replace('.log', '')
+        logger = logging.getLogger('tagcompare.%s' % name)
+
+        # Set default log levels
+        logging.getLogger("tagcompare").setLevel(logging.DEBUG)
+        if not logger.handlers:
+            if writefile:
+                logger.addHandler(self.__file_handler())
+            logger.addHandler(self.__stream_handler())
+        self._logger = logger
+
+    @property
+    def name(self):
+        return self.__name
```

Comment:
Oh yeah - fixed the test issue so it's 100% coverage now.

It's not exactly a factory though - for a factory I'd expect it to have a method like `create` which given the params it will create a new logger.  This class is the Logger itself, in that you construct a logger with logger.Logger(...).